trieste/ask_tell_optimization.py

```diff
@@ -288,7 +288,17 @@ def __init__(
                 datasets = with_local_datasets(
                     self._datasets, num_local_datasets, self._dataset_ixs
                 )
-        self._filtered_datasets = self._acquisition_rule.filter_datasets(self._models, datasets)
+            self._acquisition_rule.initialize_subspaces(search_space)
```

Comment:
We should also do this for `bayesian_optimizer.py` under `step == 1`, and add a unit test (like you did for ask_tell).

Reply:
Good spot. Done.